Utilities/Thread.cpp

```diff
@@ -1394,7 +1394,24 @@ bool handle_access_violation(u32 addr, bool is_writing, x64_context* context)
 			return true;
 		}
 
-		vm::temporary_unlock(*cpu);
+		if (cpu->id_type() == 2)
+		{
+			LOG_FATAL(MEMORY, "Access violation %s location 0x%x", is_writing ? "writing" : "reading", addr);
+
+			// TODO:
+			// RawSPU: Send appropriate interrupt
+			// SPUThread: Send sys_spu exception event
+			cpu->state += cpu_flag::dbg_pause;
+			if (cpu->check_state())
+			{
+				static_cast<spu_thread*>(cpu)->cpu_stop();
+				throw cpu_flag::exit;
```

Comment:
Need better solution than throwing. Seen it somewhere.

Reply:
Can you answer what it is? this is the review comment that needs to be solved.

Reply:
It's better to use falloc after emulation is stopped. Throwing is not portable.

Reply:
ahh, falloc may fail on non-existent blocks.
I've used it in page faults handling because its not the case there.

Reply:
May be possible to create a block though. Or use utils::memory_commit instead.

Reply:
Added a force flag for falloc, when used block is created automatically.
memory_commit had issues with memory_decommit in vm::close();